### ISSUE TITLE
make it easier to see output scene

### ISSUE THREAD

**Issue by kfarr** [3DStreet/text-to-street#2]
As a user making a text-to-street request via the python console running get_streetmix_json function inside texttostreet.py, when a response is provided I should be able to see the scene output after a few seconds.

ideas:
1) have an index.html with copy/paste destination clearly marked with comment. copy/paste the json output from the python function console output.
2) create 3dstreet json that can be downloaded, copy / pasted into a new file, or a new file is created in local filesystem that includes the new street json from open ai response as well as the original prompt in the title field.

**Comment by kfarr** [comment]
ticket for @algorush but I can't seem to assign this

**Comment by Algorush** [comment]
@kfarr hi, what if I create simple flask server with form.html to send user queries to text-to-street.py, that will opened new tab with 3dstreet scene? Is this ok for first version?

**Comment by Algorush** [comment]
But I also created index-bot.html for option 1

**Comment by kfarr** [comment]
here is the ideal case for demo purposes:

1. a request is sent from python command line (done)
2. the response from chatgpt is shown streaming in console (done, see https://github.com/Algorush/text-to-street/commit/e4303d2af5edc90fb2a2fc31ce2ff0e687e4bc43)
3. upon json being received, it opens up a new browser tab with the url of the generated scene -- needs help

I could use help with #3 

You mentioned that you already made an index.html that can copy / paste?

Maybe we can use that, and the python command-line script generates a url with the generated JSON passed directly in the URL as URL encoded data such as localhost:7001/index-bot/#streetmix-json:[URI ENCODED STREETMIX JSON HERE]

**Comment by Algorush** [comment]
> You mentioned that you already made an index.html that can copy / paste?

Yes, I slightly changed the code of the streetmix-loader component so that I could generate a scene from the ready-made json with streetmix format

**Comment by kfarr** [comment]
@Algorush great could you share that or use it do to the above idea of passing the json directly as URI data?

**Comment by Algorush** [comment]
Yes, now I’m doing just this to transmit json as URI data.
I just made a version yesterday with a simple flask server that returns generated json upon request. But it still needs to be checked first, validated before using it to generate a street. Because GPT sometimes returns invalid data in json (for example, the type of buildings that is not in the enum array). 
I came up with this option - if the generated segment type or variant is not in the json schema, then use the default value from the schema

**Comment by kfarr** [comment]
Yes that makes sense. Can you check out the streaming demo version? It’s
here

https://github.com/3DStreet/text-to-street/blob/e4303d2af5edc90fb2a2fc31ce2ff0e687e4bc43/functions/texttostreetdemo.py#L53

I couldn’t get functions working with streaming so instead it parses the
streamed output to look for a json snippet

**Comment by Algorush** [comment]
I found this (how to run function_call with streaming): https://community.openai.com/t/functions-calling-with-streaming/305742

**Comment by Algorush** [comment]
Judging by what I read about function_call, it is still better to use them when generating JSON, with them the structure of the model response is more accurate (in accordance with schema)

**Comment by kfarr** [comment]
Yeah that solution you posted looks great

**Comment by Algorush** [comment]
I'll try the streaming option later, okay? Now I have finally achieved that the street opens in a new tab upon request from the flask form

**Comment by kfarr** [comment]
congrats! do you mind checking in and sharing the current state? I could also use the index-bot.html file you mentioned

**Comment by Algorush** [comment]
> congrats! do you mind checking in and sharing the current state? I could also use the index-bot.html file you mentioned

Yes, I had created PR with it: https://github.com/3DStreet/text-to-street/pull/3

**Comment by Algorush** [comment]
the server can also accept a user request and send streetmix json. That is, we can also use streetmix-loader (adding some code) to sending a user query as an attribute. This can be helpful for future UI interface for queries
